keras/layers/core.py

```diff
@@ -16,21 +16,36 @@
 srng = RandomStreams(seed=np.random.randint(10e6))
 
 class Layer(object):
-    def __init__(self):
+    def __init__(self, name, prev):
```

Comment:
What is the purpose of passing the name of the layer and the name of the previous layer? Isn't the name of the previous layer readily accessible as `self.previous.name`?